runtime/marshal_jsonpb.go

```diff
@@ -84,16 +87,24 @@ func (j *JSONPb) marshalNonProtoField(v interface{}) ([]byte, error) {
 }
 
 // Unmarshal unmarshals JSON "data" into "v"
-// Currently it can marshal only proto.Message.
-// TODO(yugui) Support fields of primitive types in a message.
```

Comment:
I removed these lines because they are not true anymore. There are even tests that verify that unmarshaling works for non-proto types.